clients/java/src/main/java/io/camunda/zeebe/client/ZeebeClientBuilder.java

```diff
@@ -49,6 +49,13 @@ ZeebeClientBuilder applyEnvironmentVariableOverrides(
    */
   ZeebeClientBuilder gatewayAddress(String gatewayAddress);
 
+  /**
+   * @param tenantId the tenant identifier which is used for tenant-aware deployments when no tenant
+   *     identifier is set for a command. Default value is an empty String since the property isn't
+   *     allowed when multi-tenancy is disabled in the Zeebe gateway.
```

Comment:
>Default value is an empty String since the property isn't allowed when multi-tenancy is disabled in the Zeebe gateway.

❓ This makes me wonder, can we allow clients to explicitly refer to the default tenant in requests when multi-tenancy is disabled? The expected behavior would be equivalent from both Zeebe's and the client's perspectives.

Doing so would allow us to set the default tenant id to `<default>` by default. This would teach users that the identifier for the default tenant is `<default>` early on.

Reply:
:+1: I like the idea.

In that case, we would need to adjust the validation in the Gateway as well, so when multi-tenancy is disabled, only the `<default>` tenant ID is allowed.